marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx

```diff
@@ -364,25 +369,118 @@ const EntityMapTable: React.FC<Props> = (props) => {
     }
   });
 
+  const getFilteredData = (value, tree) => {
+    let filteredArray = deepCopy(tree);
+    const parser = (data) => {
+      let moreRowObj;
+      let searchStr = value.toLowerCase();
+      for (let i = 0; i < data.length; i++) {
+        let name = data[i].filterName.toLowerCase();
+        if (!name.includes(searchStr) && (!data[i].hasOwnProperty("children") || (data[i].hasOwnProperty("children") && data[i].children.length === 0))) {
+          data.splice(i, 1);
+          i--;
+        } else if (!name.includes(searchStr) && data[i].hasOwnProperty("children") && data[i].children.length > 0) {
+          parser(data[i].children);
+        } else if (name.includes(searchStr) && data[i].hasOwnProperty("parentVal") && data[i].name !== data[i].filterName) {
+          data[i].filterMatch = true;
+          let parentKey = getParentKey(data[i].key, props.entityTypeProperties);
+          moreRowObj = {key: parentKey * 10, name: "more", filterName: "more", filterMatch: false, parentVal: "", type: "", parentKey: parentKey, searchKey: data[i].key};
+        }
 
-  const getPropValueFromDataIndex = (record, index) => {
-    let res;
-    if (record.hasOwnProperty("children")) {
-      res = "-"+record[index];
-      record["children"].forEach(obj => {
-        res = res + getPropValueFromDataIndex(obj, index);
-      });
-      return res;
-    } else {
-      return "-"+record[index];
+        if (data[i] && data[i].hasOwnProperty("children") && data[i].children.length === 0) {
+          data.splice(i, 1);
+          i--;
+        }
+      }
+
+      if (moreRowObj && data.length > 0) {
+        data.push(moreRowObj);
+      }
+
+      return filteredArray;
+    };
+    return parser(filteredArray);
+  };
+
+  const getRowSiblings = (parentKey) => {
```

Comment:
Can add a few comments to hint at what's going on in some of the new logic? I'm looking at the following methods:

- getFilteredData
- *Siblings

Reply:
Added the comments. Thanks.